internal/pipeline/testdata/pkg-with-generator/configs/db/starlark-httpbin.yaml

```diff
@@ -0,0 +1,34 @@
+apiVersion: fn.kpt.dev/v1alpha1
+kind: StarlarkRun
+metadata:
+  name: httpbin-gen
+source: |
+  # add resource httpbin if doesnt exist 
+  httpbin_deployment = {
+        "apiVersion": "apps/v1",
+        "kind": "Deployment",
+        "metadata": {
+                "name": "httpbin",
+        },
+        "spec": {
+                "replicas": 4,
+                "template": {
+                        "spec": {
+                                "containers": [
+                                        {
+                                                "name": "httpbin",
+                                                "image": "kennethreitz/httpbin",
+                                                "ports": [
+                                                        {
+                                                                "containerPort": 9876
+                                                        }
+                                                ]
+                                        }
+                                ]
+                        }
+                }
+        }
+  }
+  def ensureHTTPBin(resources):
```

Comment:
You don't have to define a function if you don't need a for loop.

Reply:
Made it idempotent and also using snake-cased syntax

Reply:
That's great